Allow sprites named stage

kchadha opened the pull request:
Resolves #49 which was causing save/load failures for 2.0 and 3.0 projects.

kchadha commented:
@picklesrus and I discovered an issue with saving/loading a project from 2.0 where a sprite was named 'stage'. Turns out that we can reproduce this error (completely separate from 2.0) where naming a sprite 'stage' or 'Stage' in the 3.0 editor does not allow save/load because the sb3 fails validation. This fix will need to be deployed to scratch-projects and scratch-analysis (in addition to being updated in scratch-vm).

kchadha commented:
Hmmm for some reason commit lint is failing on travis (but running `npm run commitmsg`  locally is not failing for me. Could it be our node version? I see that travis is running node 11...

joker314 commented on lib/sb3_definitions.json:
```diff
@@ -347,8 +347,7 @@
                 "name": {
-                    "type": "string",
-                    "not": {"enum": ["Stage", "stage"]}
+                    "type": "string"
```
Should validation not fail if you try to name a sprite `_stage_`, or is that handled elsewhere?

```suggestion
                    "type": "string"
                    "not": {"enum": ["_stage_"]}
```

kchadha replied:
@joker314, good catch! It looks like both the 2.0 and 3.0 editors prevent renaming a sprite to `_stage_` so that should be included in the validation.

kchadha commented:
@thisandagain, added the fix that @joker314 suggested and added a test fixture in the invalid sub directory for it.

I also updated the commitlint package (verifying locally that it works with both node 8 and 11), so now the build is not failing on travis anymore.
